sentry-ruby/lib/sentry/redis.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+module Sentry
+  # @api private
+  class Redis
+    OP_NAME ||= "db.redis.command"
+    LOGGER_NAME ||= :redis_logger
+
+    def initialize(commands, host, port, db)
+      @commands, @host, @port, @db = commands, host, port, db
+    end
+
+    def instrument
+      return yield unless Sentry.initialized?
+
+      record_span do
+        yield.tap do
+          record_breadcrumb
+        end
+      end
+    end
+
+    private
+
+    attr_reader :commands, :host, :port, :db
+
+    def record_span
+      return yield unless (transaction = Sentry.get_current_scope.get_transaction) && transaction.sampled
+
+      transaction.start_child(op: OP_NAME, start_timestamp: Sentry.utc_now.to_f).then do |sentry_span|
```

Comment:
I use then here, which is Ruby 2.6. Or we could use yield_self (Ruby 2.5).

Do we need to support Ruby 2.4?

Reply:
@lewispb Yeah we still haven't dropped Ruby 2.4 yet. So it should be supported.
Speaking of this, @sl0thentr0py do you think we can get some statistics about our users' Ruby version breakdown?